src/snapshot.rs

```diff
@@ -52,6 +53,26 @@ pub struct StartupData {
   raw_size: int,
 }
 
+impl From<Box<[u8]>> for StartupData {
```

Comment:
You don't need this variant. 
The `From<&[u8]>` implementation can also be used with boxed slices 
(and anything else that derefs to `&[u8]` for that matter, including `Vec<u8>`).